test/Mono.Linker.Tests.Cases/DataFlow/PropertyDataFlow.cs

```diff
@@ -265,6 +267,27 @@ public void TestPropertyWithExistingAttributes ()
 				[param: DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicConstructors)]
 				set { PropertyWithExistingAttributes_Field = value; }
 			}
+
+			[UnrecognizedReflectionAccessPattern (typeof (PropertyWithIndexer), "Item.set", new Type[] { typeof (Type) }, messageCode: "IL2067")]
+			[LogDoesNotContain ("'Value passed to parameter 'index' of method 'Mono.Linker.Tests.Cases.DataFlow.PropertyDataFlow.TestAutomaticPropagationType.PropertyWithIndexer.Item.set'")]
+			public void TestPropertyWithIndexer (Type myType)
```

Comment:
Maybe there should be a second test which calls the setter with correctly annotated value - and validate that it doesn't warn. That's what the original bug is about, so we should have a test case for it.